### PR TITLE
Add more event listeners

### PR DESCRIPTION
**Pull request by bsudekum**
- `onFinishLoadingMap`
- `onStartLoadingMap`
- `onLocateUserFailed`

/cc @1ec5

### CODE REVIEW

File: ios/RCTMapboxGL/RCTMapboxGL.m [hunk 1]
```diff
@@ -469,6 +469,39 @@ - (void)handleLongPress:(UITapGestureRecognizer *)sender
     }
 }
 
+- (void)mapViewDidFinishLoadingMap:(MGLMapView *)mapView
+{
+    NSDictionary *event = @{ @"target": self.reactTag,
+                             @"src": @{
+                                     @"loaded": @YES
```
Comment (1ec5) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: Is there any point to this `loaded` property if it’s always `YES`?

File: ios/RCTMapboxGL/RCTMapboxGL.m [hunk 2]
```diff
@@ -469,6 +469,39 @@ - (void)handleLongPress:(UITapGestureRecognizer *)sender
...
+                             };
+    
+    [_eventDispatcher sendInputEventWithName:@"onStartLoadingMap" body:event];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFailToLocateUserWithError:(NSError *)error
+{
+    NSDictionary *event = @{ @"target": mapView.reactTag,
+                             @"src": @{
+                                     @"code": @(error.code),
```
Comment (1ec5) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: `code` is one of the enum values [listed here](https://developer.apple.com/library/mac/documentation/CoreLocation/Reference/CoreLocationConstantsRef/#//apple_ref/c/tdef/CLError). Are developers expected to compare the code to (say) `2` to discover a network issue, or does React Native automatically expose the `CLError` values to the JS side as well?

File: ios/RCTMapboxGL/RCTMapboxGL.m (hunk 1 above)
Comment (bsudekum) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: Yeah, these should be removed.

File: ios/RCTMapboxGL/RCTMapboxGL.m (hunk 2 above)
Comment (bsudekum) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: True, do you think `message` is enough?

Comment (1ec5) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: `description` is intended for debugging purposes; you should never compare it to a string to identify a particular condition. The wording could also potentially change from release to release. An NSError object should be identified by its `code`, but that means you may have to export the CLError values as constants, just as you did for style URLs. Also, you should also send along the error’s `localizedDescription`: this is the potentially localized error message that’s suitable for presenting to the user.

**Pull request merged**